src/connectors/youtube-music.js

```diff
@@ -26,10 +23,33 @@ Connector.albumSelector = [
 	'.ytmusic-player-bar .yt-formatted-string.style-scope.yt-simple-endpoint[href*="browse/FEmusic_library_privately_owned_release_detailb_"]',
 ];
 
+Connector.getArtistTrack = () => {
+	let artist; let track;
+	if (Connector.getAlbum()) {
+		artist = Util.getTextFromSelectors(channelNameSelector);
+		track = Util.getTextFromSelectors(trackSelector);
+	} else {
+		({ artist, track } = Util.processYtVideoTitle(Util.getTextFromSelectors(trackSelector)));
+		if (!artist) {
+			artist = Util.getTextFromSelectors(channelNameSelector);
+		}
+	}
+	return { artist, track };
+};
+
 Connector.timeInfoSelector = '.ytmusic-player-bar.time-info';
 
 Connector.isPlaying = () => {
 	return Util.getAttrFromSelectors(playButtonSelector, 'd') === playingPath;
 };
 
 Connector.isScrobblingAllowed = () => !Util.isElementVisible(adSelector);
+
+const baseStateChangedWorker = Connector.stateChangedWorker;
+Connector.stateChangedWorker = () => {
+	Connector.metadataFilter = MetadataFilter.getDefaultFilter();
+	if (! Connector.getAlbum()) {
+		Connector.applyFilter(MetadataFilter.getYoutubeFilter());
+	}
+	baseStateChangedWorker();
+};
```

Comment:
This is a bad idea, and it's relies on the internal base connector implementation.

Remember, you can use built-in filter functions. I would use the following approach:
1. Use `MetadataFilter.youtube` function to filter the track title. Since it's just a function, you can use it in a condition block.
2. Add some caching to prevent calling the filter function on every DOM change.

I know this limitation, but for now we have no cleaner solution.

Reply:
Can this be resolved given the new implementation?

Reply:
It still looks hacky, but for now we have no other solutions.